test/hotspot/jtreg/compiler/whitebox/RelocateAllNMethods.java

```diff
@@ -0,0 +1,107 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=Serial
+ * @bug 8316694
+ * @summary Relocates all nmethods and runs garbage collector to cleanup old versions
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseSerialGC compiler.whitebox.RelocateAllNMethods
+ */
+
+/*
+ * @test id=Parallel
+ * @bug 8316694
+ * @summary Relocates all nmethods and runs garbage collector to cleanup old versions
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseParallelGC compiler.whitebox.RelocateAllNMethods
+ */
+
+/*
+ * @test id=G1
+ * @bug 8316694
+ * @summary Relocates all nmethods and runs garbage collector to cleanup old versions
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseG1GC compiler.whitebox.RelocateAllNMethods
+ */
+
+/*
+ * @test id=Shenandoah
+ * @bug 8316694
+ * @summary Relocates all nmethods and runs garbage collector to cleanup old versions
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseShenandoahGC compiler.whitebox.RelocateAllNMethods
+ */
+
+/*
+ * @test id=ZGC
+ * @bug 8316694
+ * @summary Relocates all nmethods and runs garbage collector to cleanup old versions
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.misc java.management
+ *
+ * @requires vm.opt.DeoptimizeALot != true
+ *
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -XX:+SegmentedCodeCache -XX:+UseZGC compiler.whitebox.RelocateAllNMethods
+ */
+
+package compiler.whitebox;
+
+import jdk.test.whitebox.WhiteBox;
+
+public class RelocateAllNMethods {
```

Comment:
Relocate all without destination looks confusing. You need to specify where to relocate nmethods. Also there would be not many compiled code. It mostly tests relocation of code which cannot be relocated.
IMO it's better to compile a bunch of method, to relocate them, and to execute them.